### PR TITLE
fix: Move rules that depend on data from demographic table to Cohort_lookup

### PR DESCRIPTION
**Pull request by MWClayson-NHS**
## Description

Amends that of records were blocked from making it to the cohort distribution table, were not updating the participant management record. 

Update Participant will now get the current participant from the participant management table. this is just used to check if the participant already existed.

if a record reaches the cohort lookup rules its assumed that the participant will exist in the Cohort distribution table.

## Type of changes

- [ ] Refactoring (non-breaking change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [ ] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [ ] I have followed the code style of the project
- [ ] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [x] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

**Comment by stephhou** [comment]
We can't assume that an amend will already be in cohort distribution we need to handle the possibility that it returns null. This is in the case of an ADD being sent, it then has an exception, we receive an update but it is already in participant management. I'll go over the rules and see how it can be handled

File: application/CohortManager/src/Functions/ScreeningValidationService/LookupValidation/Breast_Screening_cohortRules.json [hunk 1]
```diff
@@ -29,5 +29,36 @@
         "Expression": "!(recordType AND reasonForRemoval AND postcode AND primaryCareProvider)"
       }
     ]
+  },
+  {
+    "WorkflowName": "AMENDED",
+    "Rules": [
+      {
+        "RuleName": "22.ParticipantMustExist.Fatal",
```
Comment (andrewmorris43) [inline application/CohortManager/src/Functions/ScreeningValidationService/LookupValidation/Breast_Screening_cohortRules.json:37]: Rule 22 is in both cohort and lookup rules. Is that right?

Comment (MWClayson-NHS) [reply application/CohortManager/src/Functions/ScreeningValidationService/LookupValidation/Breast_Screening_cohortRules.json:37]: Cohort will lookup to the cohort distribution tables while, the Lookup Validation will lookup with the participant management

**Comment by MWClayson-NHS** [comment]
@stephhou 
You are right,
I think we need to rethink the Cohort rules here more holistically as we should determine at that stage if its in the cohort distribution table rather than expecting it to be based on the amend / add flag. We can of course receive an amend that failed initial validation so didn't make it to distribution so the checks in the cohort_rules based on the flag should be changed.

File: application/CohortManager/src/Functions/screeningDataServices/updateParticipantDetails/updateParticipantDetails.cs [hunk 2]
```diff
@@ -45,11 +45,8 @@ public async Task<HttpResponseData> Run([HttpTrigger(AuthorizationLevel.Anonymou
                 participantCsvRecord = JsonSerializer.Deserialize<ParticipantCsvRecord>(requestBody);
             }
 
-            var existingParticipantData = GetLastAddedParticipant(participantCsvRecord);
-            if (existingParticipantData == null)
-            {
-                return _createResponse.CreateHttpResponse(HttpStatusCode.OK, req, "The participant was not added to the database because no historical record be found");
-            }
+            var existingParticipantData = _participantManagerData.GetParticipant(participantCsvRecord.Participant.NhsNumber, participantCsvRecord.Participant.ScreeningId);
```
Comment (SamRobinson75684) [inline application/CohortManager/src/Functions/screeningDataServices/updateParticipantDetails/updateParticipantDetails.cs:48]: where are you getting participant from now?

File: application/CohortManager/src/Functions/ScreeningValidationService/LookupValidation/Breast_Screening_cohortRules.json (hunk 1 above)
Comment (stephhou) [reply application/CohortManager/src/Functions/ScreeningValidationService/LookupValidation/Breast_Screening_cohortRules.json:37]: The participant doesn't need to exist in cohort distribution if it is an amend.

File: application/CohortManager/src/Functions/screeningDataServices/updateParticipantDetails/updateParticipantDetails.cs (hunk 2 above)
Comment (stephhou) [reply application/CohortManager/src/Functions/screeningDataServices/updateParticipantDetails/updateParticipantDetails.cs:48]: This one should be the participant management data as it's used for the lookup validations. At this point we just check to see if they exist in the participant management table.

**Review by SamRobinson75684 — DISMISSED** [review]

File: application/CohortManager/src/Functions/ScreeningValidationService/LookupValidation/Breast_Screening_cohortRules.json (hunk 1 above)
Comment (MWClayson-NHS) [reply application/CohortManager/src/Functions/ScreeningValidationService/LookupValidation/Breast_Screening_cohortRules.json]: @stephhou  Ill remove

**Review by andrewmorris43 — APPROVED** [review]

**Review by SamRobinson75684 — APPROVED** [review]

**Pull request merged**